lib/wovnrb.rb

```diff
@@ -85,7 +85,12 @@ def switch_lang(body, values, url, lang=@store.settings['default_lang'], headers
       lang = Lang.new(lang)
       ignore_all = false
       new_body = []
-      body.each do |b|
+
+      # generate full_body to intercept
+      full_body = ''
```

Comment:
`full_body = body.join('')` does not work?

Reply:
body is not array of string, it's a rack file iterator, only each works on it

Reply:
`full_body = body.inject(:+)` would work imo

Reply:
I tried many thins
```
NoMethodError: undefined method `inject' for #<Rack::File::Iterator:0x007f90ddc65c80>
```
Please look here for reference https://www.rubydoc.info/gems/rack/Rack/File/Iterator

Reply:
hm, that's strange because it works for me :'( Btw, I think we could stop here. Thanks for checking

Reply:
if you try with only tests it works because in the tests we use array of strings...
I tried directly on the Sinatra app and it's not working